File: frontend/src/api/adminService.js
import api from "./api";

export const getPlatformOverview = async () => {
    return await api.get("/admin/overview");
};

export const getAllUsersSummary = async () => {
    return await api.get("/admin/users");
};

export const getUserAnalytics = async (userId, timeRange = "30d") => {
    return await api.get(`/admin/users/${userId}/analytics?timeRange=${encodeURIComponent(timeRange)}`);
};

export const updateUserStatus = async (userId, status) => {
    return await api.patch(`/admin/users/${userId}/status`, { status });
};

export const updateUserRole = async (userId, role) => {
    return await api.patch(`/admin/users/${userId}/role`, { role });
};

export const deleteUser = async (userId) => {
    return await api.delete(`/admin/users/${userId}`);
};

File: backend/src/main/java/com/nirmal/momentum/service/AdminService.java
package com.nirmal.momentum.service;

import com.nirmal.momentum.common.ApiResponse;
import com.nirmal.momentum.dto.admin.*;

import java.util.List;

public interface AdminService {

    ApiResponse<AdminPlatformOverviewResponse> getPlatformOverview();

    ApiResponse<List<AdminUserSummaryResponse>> getAllUsersSummary();

    ApiResponse<AdminUserAnalyticsResponse> getUserAnalytics(Long userId, String timeRange);

    ApiResponse<Void> updateUserStatus(Long userId, UpdateUserStatusRequest request);

    ApiResponse<Void> updateUserRole(Long userId, UpdateUserRoleRequest request);

    ApiResponse<Void> deleteUser(Long userId);
}
